### DISCUSSION TITLE
How to classify randomly generated data sequence in River

### DISCUSSION THREAD

**Discussion by Emmy-Obasi** [online-ml/river#976] [Q&A] [answered]
Hi everyone.
I am working on research, which is a classification project with 4 input features, say x1, x2, x3, and x4.  The output class is either 1 or 0 depending on the pattern of inputs.  x1, to x4, is to be generated as a set of random numbers in sequence.  I hope that for each sequence, a model should classify the outcome.
**My Question:**
1.  How can I achieve this without a training set?
2. How do I achieve this task using Gaussian Naive Bayes (Maybe a link to an example)
Your kind contributions would be greatly appreciated. 

Thank you.

**Comment by MaxHalford** [comment, marked as answer]
Hello there.

> The output class is either 1 or 0 depending on the pattern of inputs.

Is there a relationship between the inputs and the outputs? What kind of data is this?

> How can I achieve this without a training set?

Well if you don't have training data, you can't train a supervised model. That includes Gaussian Naive Bayes (see River's [`GaussianNB`](https://riverml.xyz/0.11.1/api/naive-bayes/GaussianNB/)).

Without a supervised model, you have two options:

1. Start with a rule-based system, which boils down to classifying data with if/else rules. You can get sophisticated and use those rules to generate labels that you can then feed into a supervised model. There's a whole literature on the topic revolving around pseudo labeling, self-supervised learning, weak supervision, etc.
2. Use a clustering model to group your data, and then assign a class to each label. Again, once you've done, you can generate labels that you feed into a supervised model. Yet again, that's all very experimental and you'll have to get your hands dirty (but then again that's what research is about :))

I hope that helps.
